lib/mongoid/clients/factory.rb

```diff
@@ -59,12 +60,20 @@ def default
       # @since 3.0.0
       def create_client(configuration)
         raise Errors::NoClientsConfig.new unless configuration
-        if configuration[:uri]
-          Mongo::Client.new(configuration[:uri], options(configuration))
+        config = configuration.dup
+        uri = config.delete(:uri)
+        database = config.delete(:database)
+        hosts = config.delete(:hosts)
+        opts = config.delete(:options) || {}
+        unless config.empty?
+          default_logger.warn("Unknown config options detected: #{config}. Please check your config files.")
```

Comment:
The options can come from a hash, not necessarily a file, right? If so I suggest removing the second sentence and just leaving the first one.

Reply:
Right, fixed.